released/packages/coq-math-classes/coq-math-classes.8.9.1/opam

```diff
@@ -0,0 +1,44 @@
+opam-version: "2.0"
+maintainer: "b.a.w.spitters@gmail.com"
+
+homepage: "https://github.com/coq-community/math-classes"
+dev-repo: "git+https://github.com/coq-community/math-classes.git"
+bug-reports: "https://github.com/coq-community/math-classes/issues"
+license: "Public Domain"
+
+synopsis: "A library of abstract interfaces for mathematical structures in Coq."
+description: """
+Math classes is a library of abstract interfaces for mathematical structures, such as
+*  Algebraic hierarchy (groups, rings, fields, …)
+*  Relations, orders, …
+*  Categories, functors, universal algebra, …
+*  Numbers: N, Z, Q, …
+*  Operations, (shift, power, abs, …)
+It is heavily based on Coq’s new type classes in order to provide: structure inference, multiple inheritance/sharing, convenient algebraic manipulation (e.g. rewriting) and idiomatic use of notations.
+"""
+
+build: [
+  [ "./configure.sh" ]
+  [ make "-j%{jobs}%" ]
+]
+install: [make "install"]
+depends: [
+  "ocaml"
+  "coq" {(>= "8.6" & < "8.11~")}
```

Comment:
I would not add the constraint on `8.11` here as we do not know yet if it will be compatible (in short, no upper bounds).

Reply:
This discussion probably belongs elsewhere (maybe you want to open an issue). But I think having the bound is good, since it means we can use 8.10 betas to test the OPAM packages in the `released` archive. The more testing betas receive, the better. In particular all MathComp packages already use the `< "8.11~"` bound. If we allow "open upwards" bounds such as `{>= "8.6"}`, we should also allow upwards closed bounds for versions in beta.

Reply:
OK, I see your point, I will think about it.

Reply:
Opam is missing some dedicated logic to make the difference between "known to be compatible" and "could be compatible but this is not known". The same kind of discussions happens in the OCaml world, e.g. https://discuss.ocaml.org/t/experimenting-with-a-new-opam-repository-release-strategy-for-large-libraries/2918.

Reply:
OK thanks the link.